aws/resource_aws_ses_sending_ip_pool.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sesv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsSesSendingIpPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSesSendingIpPoolCreate,
+		Read:   resourceAwsSesSendingIpPoolRead,
+		Update: resourceAwsSesSendingIpPoolUpdate,
+		Delete: resourceAwsSesSendingIpPoolDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ips": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsSesSendingIpPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesv2Conn
+
+	poolName := d.Get("name").(string)
+	_, err := conn.CreateDedicatedIpPool(&sesv2.CreateDedicatedIpPoolInput{
+		PoolName: aws.String(poolName),
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating SESv2 ip pool: %s", err)
+	}
+
+	d.SetId(poolName)
+
+	// Set other properties of the sending pool
```

Comment:
we can condense L54-L58 with `return resourceAwsSesSendingIpPoolUpdate(d, meta)`